api4/config.go

```diff
@@ -431,3 +438,14 @@ func makeFilterConfigByPermission(accessType filterType) func(c *Context, struct
 		return c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionManageSystem)
 	}
 }
+
+func updateTranslations(a app.AppIface, diffs config.ConfigDiffs) error {
+	for _, diff := range diffs {
+		if diff.Path != "LocalizationSettings.DefaultServerLocale" {
```

Comment:
I'm not a fan of us using these path values for conditionals because they create areas of code that are not visible to the compiler if they underlying structs or fields are changed.

In digging into [their origin as an auditing tool](https://github.com/mattermost/mattermost-server/pull/17623) I think it made sense. Then [a sanitization mechanism was added](https://github.com/mattermost/mattermost-server/pull/18855) and @streamer45 rightly suggested that [struct tags would be an improved implementation](https://mattermost.atlassian.net/browse/MM-39604).

I wonder if we could do this conditional call to update the translations elsewhere—directly against the field names—comparing old vs new values? Something like:

```go
if oldConfig.LocalizationSettings.DefaultServerLocale != newConfig.LocalizationSettings.DefaultServerLocale {
  // update translations
}

Reply:
Yes, that's a valid point about relying on a string in this case. We have the original objects, so I changed the implementation to use those instead.